api/src/main/java/org/openmrs/module/appointmentscheduling/api/db/AppointmentStatusHistoryDAO.java

```diff
@@ -58,4 +58,14 @@
 	 * @return
 	 */
 	public void purgeHistoryBy(Appointment appointment);
+
+	/**
+	 * Returns previous status of a given appointment
+	 */
+	@Transactional(readOnly = true)
+	AppointmentStatusHistory getPreviousAppointmentStatus(Appointment appointment);

```

Comment:
I think this would be clearer if it was called getMostRecentAppointmentStatusHistory